src/Nethermind/Nethermind.Taiko/Rpc/TaikoRpcModule.cs

```diff
@@ -0,0 +1,278 @@
+
+        return origin is null ? ResultWrapper<L1Origin?>.Fail("not found") : ResultWrapper<L1Origin?>.Success(origin);
+    }
+
+    public Task<ResultWrapper<PreBuiltTxList[]?>> taikoAuth_txPoolContent(Address beneficiary, UInt256 baseFee, ulong blockMaxGasLimit,
+        ulong maxBytesPerTxList, Address[]? localAccounts, int maxTransactionsLists)
+    {
+        KeyValuePair<AddressAsKey, Queue<Transaction>>[] pendingTxs =
+            _txPoolBridge.GetPendingTransactionsBySender()
+                .ToDictionary(tx => tx.Key, tx => new Queue<Transaction>(tx.Value.Where(tx => !tx.SupportsBlobs && tx.CanPayBaseFee(baseFee))))
+                .ToArray();
+
+        if (localAccounts is not null)
+        {
+            KeyValuePair<AddressAsKey, Queue<Transaction>>[] localTxs = pendingTxs
+                .Where(txPerAddr => localAccounts.Contains(txPerAddr.Key.Value) && txPerAddr.Value.Any())
+                .ToArray();
+
+            KeyValuePair<AddressAsKey, Queue<Transaction>>[] remoteTxs = pendingTxs
+                .Where(txPerAddr => !localAccounts.Contains(txPerAddr.Key.Value) && txPerAddr.Value.Any())
+                .ToArray();
+
+            pendingTxs = [.. localTxs, .. remoteTxs];
+        }
+
+
+
+        BlockHeader? head = _blockFinder.Head?.Header;
+
+        if (pendingTxs.Length is 0 || head is null)
+        {
+            return ResultWrapper<PreBuiltTxList[]?>.Success([]);
+        }
+
+        using IReadOnlyTxProcessingScope scope = _readonlyTxProcessingEnv.Build(Keccak.EmptyTreeHash);
+
+        return ResultWrapper<PreBuiltTxList[]?>.Success(ProcessTransactions(scope.TransactionProcessor, scope.WorldState, new BlockHeader(
+                head.Hash!,
+                Keccak.OfAnEmptySequenceRlp,
+                beneficiary,
+                UInt256.Zero,
+                head!.Number + 1,
+                (long)blockMaxGasLimit,
+                head.Timestamp + 1,
+                [])
+        {
+            TotalDifficulty = 0,
+            BaseFeePerGas = baseFee,
+            StateRoot = head.StateRoot,
+            IsPostMerge = true,
+        }, pendingTxs, maxTransactionsLists, maxBytesPerTxList));
+    }
+
+
+    private readonly TxDecoder _txDecoder = Rlp.GetStreamDecoder<Transaction>() as TxDecoder ?? throw new NullReferenceException(nameof(_txDecoder));
+
+    public PreBuiltTxList[] ProcessTransactions(ITransactionProcessor txProcessor, IWorldState worldState, BlockHeader blockHeader, KeyValuePair<AddressAsKey, Queue<Transaction>>[] txSource, int maxBatchCount, ulong maxBytesPerTxList)
+    {
+        lock (worldState)
+        {
+            if (txSource.Length is 0 || blockHeader.StateRoot is null)
+            {
+                return [];
+            }
+
+            List<PreBuiltTxList> Batches = [];
+
+            List<Transaction> currentBatch = [];
+
+            void CommitBatch()
+            {
+                byte[] list = EncodeAndCompress(currentBatch.ToArray());
```

Comment:
Not really used apart from length? Then allocating array is not needed and we can just pass length from the method?